extensions/amp-ad/0.1/amp-ad-api-handler.js

```diff
@@ -153,6 +153,15 @@ export class AmpAdApiHandler {
           {requestedHeight: height, requestedWidth: width},
           targetOrigin,
           this.is3p);
+    }).catch(() => {
+      const targetOrigin =
+          this.iframe_.src ? parseUrl(this.iframe_.src).origin : '*';
+      postMessage(
+          this.iframe_,
+          'embed-size-denied',
+          {requestedHeight: height, requestedWidth: width},
+          targetOrigin,
+          this.is3p_);
```

Comment:
its just a single line of code, so definitely not a lot of duplication. I don't think a helper helps a lot here.